client-java/doc/flow

```diff
@@ -0,0 +1,44 @@
+client
+  connects to server
+server
+  sends CONNECTED
+client
+  sends SUBSCRIBE with token
+server
+  associates websocket context with token
+  sends SUBSCRIBED
+client
+  waits
+server
+  reads resource from topic
+  retrieves websocket session via token
```

Comment:
Good point. yeah that could be a point of confusion.

Reply:
You cannot keep a running thread on the server per websocket, that would eat through resources pretty quickly. The Web socket sessions have to be stored in a cache that an incoming request can look up based upon the token. The session itself would be store in a weak hashmap, so if the connection dies, the cache would let it be GC'd. and the actuial entry would be removed from the map.

You would keep a `Set<String>` of active tokens alongside a WeakHashMap<String,WeakReference<WebSocketSession>>. This means that when there are no more resources for the token, it can be removed from the set. This would mean no more references for the token, meaning the entry in the map would be removed. On the other hand, if the socket connection is closed, the session can still be garbage collected, freeing memory.

I hope this makes sense.

Reply:
I think functionally we at least require a pool-of-resources-per-client (kafka consumer with offset, websocket and its state, etc.)
The filtered-resource-service can be most-easily scaled, the majority of running threads will be waiting, we're interested in low-latency-per-resource, and threads don't have a particularly high overhead on any modern machine.
While we *could* do it with some round-robin-ing and big messy WeakRef state storage, I feel going for the simpler "allocate a new running thread and all the resources it requires in its lifetime" is much simpler and may be more performant in the areas we're interested in.